src/components/Search/Input.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useRef, useEffect } from 'react';
+
+interface Props {
+  onUpdateWord: (value: string) => void,
+  word: string,
+}
+
+const Input: React.FC<Props> = function Input({
```

Comment:
Typing React FC components using React.FC it's ok but discouraged. Actually  there is a PR where they are planning to remove it.

It's better to type the props directly 

`const Input = function Input({ onUpdateWord, word }: Props) {}`

See: https://github.com/facebook/create-react-app/pull/8177 
Also see: https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript